control/plugin_manager.go

```diff
@@ -714,7 +716,7 @@ func (p *pluginManager) UnloadPlugin(pl core.Plugin) (*loadedPlugin, serror.Snap
 	p.loadedPlugins.remove(plugin.Key())
 
 	// Remove any metrics from the catalog if this was a collector
-	if plugin.TypeName() == "collector" {
+	if plugin.TypeName() == "collector" || plugin.TypeName() == core.StreamingCollectorPluginType.String() {
```

Comment:
I would like to keep the same pattern for "collector" as well and change it to
```
if plugin.TypeName() == core.CollectorPluginType.String() || plugin.TypeName() == core.StreamingCollectorPluginType.String() {
```

but it's a minor thing - @jcooklin, I will prepare PR with that to your repo - accept it or reject, up to you